src/android/java/io/jxcore/node/BtToSocketBase.java

```diff
@@ -0,0 +1,147 @@
+        return localHostSocket == null || localHostSocket.getInetAddress() == null ? null : localHostSocket.getInetAddress().toString();
+    }
+
+    public void Stop() {
+        mStopped = true;
+
+        StreamCopyingThread tmpSCTrec = ReceivingThread;
+        ReceivingThread = null;
+        if(tmpSCTrec != null){
+            print_debug("BtToSocketBase","Stop ReceivingThread");
+            tmpSCTrec.Stop();
+        }
+
+        StreamCopyingThread tmpSCTsend = SendingThread;
+        SendingThread = null;
+        if(tmpSCTsend != null){
+            print_debug("BtToSocketBase","Stop SendingThread");
+            tmpSCTsend.Stop();
+        }
+
+        CloseSocketAndStreams();
+
+        if (mmInStream != null) {
+            try {print_debug("BtToSocketBase","Close bt in");
+                mmInStream.close();} catch (IOException e) {print_debug("BtToSocketBase","Close error : " + e.toString());}
+        }
+
+        if (mmOutStream != null) {
+            try {print_debug("BtToSocketBase","Close bt out");
+                mmOutStream.close();} catch (IOException e) {print_debug("BtToSocketBase","Close error : " + e.toString());}
+        }
+
+        if (mmSocket != null) {
+            try {print_debug("BtToSocketBase","Close bt socket");
+                mmSocket.close();} catch (IOException e) {print_debug("BtToSocketBase","Close error : " + e.toString());}
+        }
+    }
+
+    protected  void CloseSocketAndStreams() {
+
+        InputStream tmpLocStrIn = LocalInputStream;
+        LocalInputStream = null;
+        if (tmpLocStrIn != null) {
+            try {print_debug("BtToSocketBase","Close local in");
+                tmpLocStrIn.close();} catch (Exception e) {
+                print_debug("BtToSocketBase","Close error : " + e.toString());
+            }
+        }
+
+        OutputStream tmpLocStrOut = LocalOutputStream;
+        LocalOutputStream = null;
+        if (tmpLocStrOut != null) {
+            try {print_debug("BtToSocketBase","Close LocalOutputStream");
+                tmpLocStrOut.close();} catch (Exception e) {
+                print_debug("BtToSocketBase","Close error : " + e.toString());
+            }
+        }
+
+        Socket tmpLHSoc = localHostSocket;
+        localHostSocket = null;
+        if (tmpLHSoc != null) {
+            try {print_debug("BtToSocketBase","Close localHostSocket");
+                tmpLHSoc.close();} catch (Exception e) {
+                print_debug("BtToSocketBase","Close error : " + e.toString());
+            }
+        }
+    }
+    public void SetIdAddressAndName(String peerId,String peerName,String peerAddress) {
+        mPeerId = peerId;
+        mPeerName = peerName;
+        mPeerAddress = peerAddress;
+    }
+    public String GetPeerId() {
+        return mPeerId;
+    }
+    public String GetPeerName(){return mPeerName;}
+    public String GetPeerAddress(){return mPeerAddress;}
+
+    protected  void print_debug(String who,String message){
```

Comment:
I don't understand why this function exists. Why can't we just use Log.i everywhere?